Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -525,6 +526,9 @@ export class ResourceGroup {
                 console.log(tl.loc("LogDeploymentName", this.taskParameters.deploymentName));
                 armClient.deployments.createOrUpdate(this.taskParameters.resourceGroupName, this.taskParameters.deploymentName, deployment, (error, result, request, response) => {
                     if (error) {
+                        if(error.code == "ResourceGroupNotFound" && retryCount > 0){
```

Comment:
I think this change should be at the ARM sdk level in the 'Common' folder, so that if someone else uses the sdk, they don;t run into the same issue.

Reply:
Agreed but as we saw multiple cases that user gets into this issue mostly because Resource Group is in Deleting state so if we add this logic in common sdk then unnecessarily we are adding retries which eventually increase the time for task failure. Rather in our case we know that ARG Task is prime contender for facing this issue as it is deploying the template right after resource group creation.
Also, this issue is reported for the first time.
Keeping in  mind both the things I preferred to add this logic in ARG Task.
Suggestions ?

Reply:
I think it makes sense to retry in this class